File: pmvs10b-lab6-grimb000/NumberGuess/app/src/main/java/com/zizto/numberguess/LoginActivity.java
package com.zizto.numberguess;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Debug;
import android.util.Log;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class LoginActivity extends AppCompatActivity {

    EditText etEmail, etPassword;
    Button btnLogin;
    SharedPreferences userProfile;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        etEmail = findViewById(R.id.etEmail);
        etPassword = findViewById(R.id.etPassword);
        btnLogin = findViewById(R.id.btnLogin);

        userProfile = getSharedPreferences("user_profile", MODE_PRIVATE);

        btnLogin.setOnClickListener(v -> {
            String email = etEmail.getText().toString();
            String password = etPassword.getText().toString();

            if (email.isEmpty() || password.isEmpty()) {
                Toast.makeText(this, "Введите email и пароль", Toast.LENGTH_SHORT).show();
                return;
            }

            String savedEmail = userProfile.getString("email", null);
            Log.d("TAG", "------------------------------------------------------- " + savedEmail + ": " + userProfile.getString("password", ""));
            if (savedEmail == null) {
                userProfile.edit()
                        .putString("email", email)
                        .putString("password", password)
                        .apply();
                Toast.makeText(this, "Пользователь зарегистрирован", Toast.LENGTH_SHORT).show();
                goToMain();
            } else if (savedEmail.equals(email) && userProfile.getString("password", "").equals(password)) {
                Toast.makeText(this, "Вход выполнен", Toast.LENGTH_SHORT).show();
                goToMain();
            } else {
                Toast.makeText(this, "Неверный email или пароль", Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void goToMain() {

        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
        startActivity(intent);
        finish();
    }
}

File: pmvs10a-lab6-grimb000/Task73/app/src/main/java/com/example/task73/LoginActivity.kt
package com.example.task73

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.launch

class LoginActivity : AppCompatActivity() {
    private lateinit var repository: GameRepository
    private lateinit var emailInput: EditText
    private lateinit var passwordInput: EditText
    private lateinit var profileSummaryView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        repository = GameRepository(applicationContext)
        emailInput = findViewById(R.id.emailInput)
        passwordInput = findViewById(R.id.passwordInput)
        profileSummaryView = findViewById(R.id.profileSummaryView)

        findViewById<Button>(R.id.saveProfileButton).setOnClickListener {
            lifecycleScope.launch { saveProfile() }
        }
        findViewById<Button>(R.id.clearProfileButton).setOnClickListener {
            lifecycleScope.launch { clearProfile() }
        }

        lifecycleScope.launch {
            loadProfile()
        }
    }

    private suspend fun loadProfile() {
        val profile = repository.loadProfile()
        emailInput.setText(profile.email)
        passwordInput.setText(profile.password)
        profileSummaryView.text = if (profile.email.isBlank()) {
            getString(R.string.profile_summary_empty)
        } else {
            getString(R.string.profile_summary_template, profile.email, profile.password)
        }
    }

    private suspend fun saveProfile() {
        val email = emailInput.text.toString().trim()
        val password = passwordInput.text.toString().trim()

        if (email.isBlank() || password.isBlank()) {
            Toast.makeText(this, R.string.fill_profile_fields, Toast.LENGTH_SHORT).show()
            return
        }

        repository.saveProfile(UserProfile(email, password))
        loadProfile()
        Toast.makeText(this, R.string.profile_saved, Toast.LENGTH_SHORT).show()
    }

    private suspend fun clearProfile() {
        repository.clearProfile()
        loadProfile()
        Toast.makeText(this, R.string.profile_cleared, Toast.LENGTH_SHORT).show()
    }
}
